### PR TITLE
Convert BUGBOT review rules into per-directory AGENTS.md standards

### PR DESCRIPTION
**Pull request by makalaaneesh**
> Stacked on #3757. Base is `improve-branch-review`; review that one first.

### Describe the changes in this pull request

The `.cursor/BUGBOT.md` files already held coding guidance, not review guidance. But only Cursor Bugbot reads that filename, so the rules reached an author only after the code existed. All 17 scopes are now an `AGENTS.md` in the directory it governs, which coding agents load before the first edit.

- **Layout** — each scope gets `AGENTS.md` plus a `CLAUDE.md` symlink and a `.cursor/BUGBOT.md` symlink. Claude Code reads `CLAUDE.md`, not `AGENTS.md`. Bugbot reads only `.cursor/BUGBOT.md` and ignores everything else. Edit `AGENTS.md`; never edit a symlink.
- **Why symlinks and not generated files** — nothing is generated, so there is no build step and nothing that can drift.
- **Why it reaches authors now** — nested `AGENTS.md` is native in Cursor, Codex, Aider and Zed. Claude Code loads a nested `CLAUDE.md` when it reads a file in that directory.
- **Rule text** — moved verbatim. Each file gains a scope header, because some tools read only the nearest file in the tree. Titles change from "Review Rules" to "Engineering Standards".
- **Carried over from #3757** — that PR's five new BUGBOT sections are folded into the matching `AGENTS.md` files. Its two new scopes, `debezium-server-voyager/` and `src/testlivemigration/`, are converted to the same layout.
- **Root dedupe** — the product-flow list, the source-heterogeneity paragraph and the upgrade-compatibility section existed in both the root `AGENTS.md` and the root BUGBOT file. They are now single-source. The package enumeration that restated `src/AGENTS.md` is trimmed to entry points and cross-package invariants.
- **New scope** — the metric catalogue moves to `yb-voyager/src/metrics/AGENTS.md`, so it loads on demand instead of in every session.
- **Comments** — the directive is now a default of none, plus a gate. A comment has to let you name the specific wrong conclusion a reader would draw without it. Packages that mandate specific comments still win locally.
- **Skills** — `branch-review` and `post-pr-review` walk `AGENTS.md`. They fall back to the old filenames for branches cut before this change.
- **Not in this PR** — per-rule severity markers, and moving the mechanically checkable rules into golangci-lint.

### Describe if there are any user-facing changes
No user-facing changes.

### How was this pull request tested?
No code changed. Verified that all 34 symlinks resolve and are committed as mode `120000`. Verified that every `go:embed` in the tree names an explicit path, so no new markdown can be swept into a binary.

### Does your PR have changes in callhome/yugabyted payloads? If so, is the payload version incremented?
No.

### Does your PR have changes to on-disk structures that can cause upgrade issues?
No.

---

#### On-disk structures potentially causing breaking changes:
| Component        
| :----------------------------------------------: | 
| MetaDB                                           |
| Name registry json                               |
| Data File Descriptor Json                        |
| Export Snapshot Status Json                      |
| Callhome Json                                    |
| Export Status Json                               |
| YugabyteD Tables                                 |
| TargetDB Metadata Tables                         |
| Schema Dump                                      |
| AssessmentDB                                     |
| Migration Assessment Report Json                 |
| Import Data State                                |
| Export and import data queue                     |
| Data .sql files of tables                        |

### CODE REVIEW

File: yb-voyager/src/AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# src/ Shared Engineering Standards
+
+> **Scope:** files under `yb-voyager/src/`. These are standards for **writing** code here, not only for reviewing it.
+> Repo-wide standards live in `AGENTS.md` at each parent directory up to the repo root — read those as well.
+
+## Package Directory
+
+When adding new code, place it in the appropriate package:
```
Comment (makalaaneesh) [inline yb-voyager/src/AGENTS.md:8]: [AI] **Package Directory omits 7 real packages under `src/` (including `metrics/`, which has its own AGENTS.md).**

This section introduces itself as authoritative — "When adding new code, place it in the appropriate package:" — and the root `AGENTS.md` (line 90) explicitly tells readers this file "holds the full package directory and the rule for where new code belongs." But it lists 27 packages while `yb-voyager/src/` actually contains 34+ directories with Go code. The following real packages are missing:

- `metrics/` — 8 Go files. This one is especially load-bearing here because this PR also adds `yb-voyager/src/metrics/AGENTS.md` (referenced twice from root `AGENTS.md` at lines 96 and 215). A directory that has its own engineering-standards file should appear in the shared directory.
- `schemadiff/` — 13 Go files.
- `schemasnapshot/` — 12 Go files.
- `testlivemigration/` — 16 Go files.
- `archivechanges/` — 3 Go files (backs `archive changes`).
- `export/` — 1 Go file.
- `ux/` — 3 Go files.

Failure case: an author following this list to place a new schema-diffing helper, a new metric, or a new archive-changes subcommand finds no entry for `schemadiff`, `metrics`, or `archivechanges` and either invents a new package or drops code into `utils/` — which the "Package Layering" rule in `yb-voyager/AGENTS.md` explicitly discourages. The point of promoting rules from BUGBOT.md to AGENTS.md is that authors read them before writing code, so a materially incomplete canonical list here is exactly the kind of drift the new standards regime should not ship with.

Suggested fix: add rows for all seven packages above (a one-line description each is enough), or reword the intro so the list is explicitly a curated highlight rather than the "full package directory" root `AGENTS.md` promises.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by makalaaneesh — COMMENTED** [review]
[AI] Automated review — 1 finding (0 critical, 1 warning). See inline comment.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: yb-voyager/src/AGENTS.md (hunk 1 above)
Comment (makalaaneesh) [reply yb-voyager/src/AGENTS.md:8]: out of scope for this PR; will take it up separately.

File: ai/skills/branch-review/SKILL.md [hunk 2]
```diff
@@ -125,10 +126,10 @@ For each changed file, take the union of all `BUGBOT.md` / `.cursor/BUGBOT.md` f
...
+| **Tests** | New logic has tests that actually assert behavior; edge cases covered. **Test quality**: no fixed sleeps or timing-dependent assertions; count assertions on asynchronous work need justified bounds on *both* sides (a vacuous lower bound like `>= 0` asserts nothing; a missing upper bound misses over-triggering); no reads of state a concurrent process is still writing. **Coverage variants** the repo's `AGENTS.md` standards require (e.g. case-sensitive identifiers, partitioned tables, expression indexes) must be checked explicitly, not assumed. |
 | **Documentation** | Public APIs documented; non-obvious gating/branching logic commented with *why* and *when it applies*; genuinely unclear concepts (new fields, enums, labels) explained — if you can't tell what a field is for, ask. |
 
 Severity is not tied to lens: a hot-path regression or an inverted condition is Critical/Warning, not a nitpick. A concrete violation of a written BUGBOT rule defaults to **Warning**, not Suggestion. Design, scope, and clarity concerns that need author input but aren't defects go in the **Question** class (Step 5) — surface them, but don't invent a "bug" to justify them.
```
Comment (makalaaneesh) [inline ai/skills/branch-review/SKILL.md]: [AI][Suggestion] **Stale "BUGBOT rule" reference missed by the rename sweep**

Every other mention of "BUGBOT rule / BUGBOT.md" in this file was rewritten to "AGENTS.md standard / AGENTS.md" as part of this PR, but this line still says *"A concrete violation of a written **BUGBOT rule** defaults to **Warning**, not Suggestion."* Note that line 48's mentions of `BUGBOT.md` are intentional (they name the fallback filenames) — only this one is stale.

This is the exact failure mode the PR itself codifies in the newly-added Comments guideline in root `AGENTS.md`: *"When a refactor renames or restructures something, sweep the surrounding comments and error messages for the old names."*

Suggested fix:

```md
Severity is not tied to lens: a hot-path regression or an inverted condition is Critical/Warning, not a nitpick. A concrete violation of a written `AGENTS.md` standard defaults to **Warning**, not Suggestion. Design, scope, and clarity concerns that need author input but aren't defects go in the **Question** class (Step 5) — surface them, but don't invent a "bug" to justify them.
```

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by makalaaneesh — COMMENTED** [review]
[AI] Automated re-review (incremental) — 1 finding (0 critical, 0 warning, 1 suggestion). The previously-flagged root `src/AGENTS.md` package-directory gap is unchanged, but was resolved as out-of-scope by the author, so it is not re-reported. All 34 symlinks resolve cleanly and the two new AGENTS.md files (debezium-server-voyager/, src/testlivemigration/) match their prior `.cursor/BUGBOT.md` sources verbatim.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: ai/skills/branch-review/SKILL.md (hunk 2 above)
Comment (makalaaneesh) [reply ai/skills/branch-review/SKILL.md]: Fixed in e2d5475d7 — applied the suggested wording verbatim. Re-swept `ai/skills/` and the root `AGENTS.md` afterwards: the only remaining BUGBOT references are line 48 here (the fallback filenames, intentional as you noted) and the root `AGENTS.md` line documenting the symlink layout.

File: AGENTS.md [hunk 3]
```diff
@@ -99,47 +102,127 @@ Test infrastructure for Go integration tests lives in `yb-voyager/test/container
...
+## Generic coding practices
+
+- Keep code simple. Use early returns to reduce nesting. Prefer flat `if err != nil { return err }` over deeply nested success paths.
+- Prefer letting the database do set-oriented work (ordering, aggregation, grouping, deduplication) in SQL rather than post-processing rows in Go, when it does not hurt readability.
+- Use self-describing variable and function names. Avoid `rec1`/`rec2` — prefer `recCombined`/`recSharded`.
+- Remove dead code, unused functions, and leftover debugging artifacts before merging.
+- Consolidate duplicate logic into shared helpers rather than copy-pasting across switch cases or source-type implementations.
+- Anything fed into a hash, fingerprint, or serialized key must be built in a deterministic order (sort map keys and column lists first).
+
+## Comments
```
Comment (ShivanshGahlot) [inline AGENTS.md:194]: This makes sense

File: AGENTS.md
Comment (ShivanshGahlot) [inline AGENTS.md]: Claude flagged this title. We can name it like the other files.

**Review by ShivanshGahlot — APPROVED** [review]
Overall LGTM. It's mostly shifting files around so didn't require much review, verified with claude too that the files are identical to the previous files. Had a look at the additions in the root Agents.md and it makes sense.

File: AGENTS.md
Comment (makalaaneesh) [reply AGENTS.md]: Done in 3e8f73e70 — now `# YugabyteDB Voyager — Engineering Standards`, matching the `<name> — Engineering Standards` form the other scopes use. The old `# INSTRUCTIONS.md` was a stale filename from before the rename; nothing in the repo referenced it.

I also extended the subtitle to say this file is the root of the standards hierarchy, since it carries more than standards alone (build/test instructions, environment requirements, gotchas) — the title alone would otherwise under-describe it.

File: AGENTS.md (hunk 3 above)
Comment (makalaaneesh) [reply AGENTS.md:194]: picked it up from the yb-stress repo :)
